app/models/polling/scheduler.rb

```diff
@@ -0,0 +1,41 @@
+require 'rufus-scheduler'
+class Polling::Scheduler
+  attr_accessor :rufus_scheduler
+  attr_accessor :job
+  DEFAULT_TIMEOUT = '2.0s'.freeze
+
+  def initialize(opts = { overlap: false, timeout: DEFAULT_TIMEOUT })
+    @opts = opts
+    @rufus_scheduler ||= Rufus::Scheduler.new
+    @rufus_scheduler.stderr = StringIO.new
+  end
+
+  def mode(mode, time = Rails.configuration.scheduler_polling_interval)
+  # @mode is any of kind of jobs i.e in, at, every, interval and cron jobs
+    @mode = mode
+    @time = time
+    self
+  end
+
+  def perform(action = -> {})
+    @job = @rufus_scheduler.method("schedule_#{@mode}")
+                           .call(@time, **@opts, &action)
```

Comment:
is there an option to stop after x attempts and log an error? to avoid running forever.

Reply:
I have added a no of time run as a default parameter `times: DEFAULT_NUMBER_POLLS`